### PR TITLE
Storage client redesign 7762

### PR DESCRIPTION
**Pull request by IlyaFaer**
Add new method to client. From redesign items list: "New method: client.list_blobs(bucket_or_name), which functions the same as Bucket.list_blobs()"
https://github.com/googleapis/google-cloud-python/issues/7762

### CODE REVIEW

File: storage/google/cloud/storage/client.py [hunk 1]
```diff
@@ -153,6 +153,26 @@ def _pop_batch(self):
         """
         return self._batch_stack.pop()
 
+    def _use_or_init_bucket(self, bucket_or_name):
```
Comment (IlyaFaer) [inline storage/google/cloud/storage/client.py]: Added helper - for now it is used in 3 places

File: storage/google/cloud/storage/client.py [hunk 2]
```diff
@@ -394,6 +404,82 @@ def download_blob_to_file(self, blob_or_uri, file_obj, start=None, end=None):
...
+        return bucket.list_blobs(
+            max_results=max_results,
+            page_token=page_token,
+            prefix=prefix,
+            delimiter=delimiter,
+            versions=versions,
+            projection=projection,
+            fields=fields,
+            client=self
+        )
```
Comment (IlyaFaer) [inline storage/google/cloud/storage/client.py]: Proxy to bucket method

File: storage/tests/unit/test_bucket.py [hunk 3]
```diff
@@ -782,19 +782,6 @@ def test_list_blobs_w_all_arguments_and_user_project(self):
...
-        connection = _Connection({"items": []})
-        client = _Client(connection)
-        bucket = self._make_one(client=client, name=NAME)
-        iterator = bucket.list_blobs()
-        blobs = list(iterator)
-        self.assertEqual(blobs, [])
-        kw, = connection._requested
-        self.assertEqual(kw["method"], "GET")
-        self.assertEqual(kw["path"], "/b/%s/o" % NAME)
-        self.assertEqual(kw["query_params"], {"projection": "noAcl"})
```
Comment (IlyaFaer) [inline storage/tests/unit/test_bucket.py]: This is full copy of test named "test_list_blobs_defaults"

**Comment by mf2199** [comment]
I only see changes made to the client class, but they also requested to update Bucket and Blob classes.

**Comment by mf2199** [comment]
They also request to deprecate some methods, which usually implies to set up deprecation warnings.

**Pull request closed without merging**